core/arch/interp.c

```diff
@@ -3377,13 +3381,15 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
             bb->instr_start = bb->cur_pc;
             if (bb->full_decode) {
                 /* only going through this do loop once! */
-                bb->cur_pc = decode(dcontext, bb->cur_pc, bb->instr);
+                bb->cur_pc = IF_AARCH64_ELSE(decode_with_ldstex, decode)
```

Comment:
> (One reason for not decoding first: it might be worth searching for LDXR/STXR with a bit mask to avoid the cost of decoding, or caching decoded instructions, because there's a performance problem here.)

This is what fast decoding is for.  On x86 there is a fast decoder that only figures out the length and whether it's an "interesting" instruction.

If this bundle approach gets generalized to aarch32 it should be revisited where it lives and should likely be moved to block building and be closer to the existing level 0 bundling.  However, with no support for instrumentation, it's not clear how long a shelf life this will have.